### DISCUSSION TITLE
How to delete the last empty lines of a file in normal mode?

### DISCUSSION THREAD

**Discussion by OpenFoam-User** [helix-editor/helix#4431] [Q&A]
Sorry if this is a stupid question but I have searched the documentation, the discussion on this repository but I can't find an answer (even on google there is no useful results yet).

Q: How can I delete empty lines in helix when in normal mode?

I know about `d` key but it doesn't get rid of the empty line.

Thank you

**Comment by geometryolife** [comment]
`d`

**Reply by OpenFoam-User** [reply]
This doesn't delete lines as in neovim.

**Reply by dead10ck** [reply]
I'm not really sure what answer you're looking for. `d` deletes empty lines. In neovim this is `dd`. Is the extra `d` what makes it "neovimmy"?

**Reply by OpenFoam-User** [reply]
@geometryolife  I noticed this behavior in many replies (for other users as well) where some people reply with only single characters. This is very disrespectful and it is not acceptable.

**Reply by dead10ck** [reply]
If `d` isn't working for you, then the line probably isn't empty. In which case you want `xd`.

**Reply by OpenFoam-User** [reply]
@dead10ck : I want to get rid of the entire line. I don't know how to explain further but you can try it in both hx and neovim to see the difference. In neovim it really gets rid of the entire line and the cursor moves up to the previous line while in hx the cursor is still in the line that is supposed to be deleted.

**Reply by geometryolife** [reply]
> @geometryolife I noticed this behavior in many replies (for other users as well) where some people reply with only single characters. This is very disrespectful and it is not acceptable.

Sorry, thanks for pointing out my shortcomings.

**Reply by dead10ck** [reply]
> @geometryolife I noticed this behavior in many replies (for other users as well) where some people reply with only single characters. This is very disrespectful and it is not acceptable.

There's really nothing disrespectful about it at all. You asked a question and they answered it correctly.

As I said, if `d` isn't working for you, then the line probably isn't empty, it probably has nothing but whitespace in it. If you asked a question you didn't intend to, that is your mistake, not theirs.

**Reply by OpenFoam-User** [reply]
no wrong! it turns out that helix is not able to delete the last lines in a buffer when the cursor is placed in the very last line so `d` has no effect. So it is not my mistake.

**Reply by dead10ck** [reply]
You are putting the cursor on the end of the buffer, not the last line. You may notice how on the left, there is no line number, just a `~`. It means your cursor is past all of the text in the buffer. So again, you are not talking about an empty line.
